in_toto/verifylib.py

```diff
@@ -1043,45 +1044,136 @@ def reduce_chain_links(chain_link_dict):
+    by an equivalent link object.
+
+  <Arguments>
+    layout:
+            The layout specified by the project owner.
+
+    chain_link_dict:
+            A dictionary of key-link pair with step names as keys. For each
+            step name, there are one or more keyids and corresponding
+            link objects.
+
+  <Exceptions>
+    raises an Exception if verification of the delegated step fails.
+
+  <Side Effects>
+    None.
+
+  <Returns>
+    A dictionary of key-link pair with step names as keys. For each
+    step name, there are one or more keyids and corresponding
+    link objects.
+  """
+
+  for step_name, key_link_dict in six.iteritems(chain_link_dict):
+
+    for keyid, link in six.iteritems(key_link_dict):
+
+      # Extract keyid and link from the passed chain_link_dict
+      # Check if the link object is of type 'layout'
+      if link._type == "layout":
+        layout_key_dict = {}
+
+        # Retrieve the entire key object for the keyid
+        # corresponding to the link
+        layout_key_dict = {keyid: layout.keys.get(keyid)}
+
+        # Make a recursive call to in_toto_verify with the
+        # layout and the extracted key object
+        summary_link = in_toto_verify(link, layout_key_dict)
 
-def in_toto_verify(layout_path, layout_key_paths):
+        # Replace the layout object in the passed chain_link_dict
+        # with the link file returned by in-toto-verify
+        key_link_dict[keyid] = summary_link
+
+  return chain_link_dict
+
+def get_summary_link(layout, reduced_chain_link_dict):
+  """
+  <Purpose>
+    Merges the materials of the first step (as mentioned in the layout)
+    and the products of the last step and returns a new link.
+    This link reports the materials and products and summarizes the
+    overall software supply chain.
+
+  <Arguments>
+    layout:
+            The layout specified by the project owner.
+
+    reduced_chain_link_dict:
+            A dictionary of step-link pair with step names as keys.
+
+  <Exceptions>
+    None.
+
+  <Side Effects>
+    None.
+
+  <Returns>
+    A link which summarizes the materials and products of the overall
+    software supply chain.
+  """
+
+  # Create a new link object to return
+  summary_link = in_toto.models.link.Link()
+
+  first_step_link = reduced_chain_link_dict[layout.steps[0].name]
+  last_step_link = reduced_chain_link_dict[layout.steps[-1].name]
+
+  # Copy _type, name and materials from the first step link
+  summary_link.materials = first_step_link.materials
+  summary_link._type = first_step_link._type
+  summary_link.name = first_step_link.name
+
+  # Copy products, byproducts and command from the last step link
+  # as these properties would be used by the subsequent steps
+  summary_link.products = last_step_link.products
+  summary_link.byproducts = last_step_link.byproducts
+  summary_link.command = last_step_link.command
+  summary_link.return_value = last_step_link.return_value
+
```

Comment:
[Ping](https://github.com/in-toto/in-toto/pull/86/files/97c28e84fa3929b5165657a0df9123aab89ea719#r114184488)